src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1121,7 +1122,16 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
                 }
 
                 // Update the selection appropriately
-                if (shiftEnabled && _terminal->IsSelectionActive())
+                if (ctrlEnabled && multiClickMapper == 1)
+                {
+                    // Control+Click: if the text selected is a hyperlink, open the link
+                    const auto uri = _terminal->GetHyperlinkAtPosition(terminalPosition);
+                    if (!uri.empty())
+                    {
+                        _HyperlinkHandler(uri);
+                    }
+                }
+                else if (shiftEnabled && _terminal->IsSelectionActive())

```

Comment:
If there's _no_ URI at the location, and ctrl is pressed, shouldn't we also fall through to the "update the selection" case?